crates/aptos-rosetta/src/construction.rs

```diff
@@ -246,9 +246,26 @@ async fn construction_metadata(
         return Err(ApiError::ChainIdMismatch);
     }
 
+    let sequence_number = {
+        let response_account = response.inner();
+        let mut accounts = server_context.accounts.lock().await;
+        if let Some(stored_sequence_number) = accounts.get_mut(&address) {
+            // If we missed a sequence number update, we should update
+            if response_account.sequence_number > *stored_sequence_number {
+                *stored_sequence_number = response_account.sequence_number;
+            }
+            let seq_num = *stored_sequence_number;
+            *stored_sequence_number += 1;
+            seq_num
+        } else {
+            accounts.insert(address, response_account.sequence_number + 1);
+            response_account.sequence_number
+        }
+    };
+
     Ok(ConstructionMetadataResponse {
         metadata: ConstructionMetadata {
-            sequence_number: response.inner().sequence_number,
+            sequence_number,
```

Comment:
Is it intentional that seq num is now + 1 when we build this compared to before? This change wasn't mentioned in the PR summary so I just wanna confirm.

Reply:
So, the sequence number has to increase locally, because the tester doesn't handle failures due to sequence numbers being out of date and retrying the API.